src/sentry/models/event.py

```diff
@@ -89,6 +89,8 @@ def generate_node_id(cls, project_id, event_id):
     @property
     def group(self):
         from sentry.models import Group
+        if not self.group_id:
+            return None
```

Comment:
Do consumers of this property check for nulls?

Reply:
No, not yet. That is my next step since as of this PR there is no code path to create an event without a group so we cannot be in a condition where this will return None yet.

Reply:
I don't love this but I don't have any better ideas.

Reply:
As above, I think the only way to avoid this is to split events from transactions/traces. But that opens a different conceptual problem. In a world where the event grouping of choice is the incident, what's the future of groups ? Are they going to still exist ?